core/src/stores/store.rs

```diff
@@ -267,23 +269,52 @@ pub trait Store {
         &self,
         project: &Project,
         data_source_id: &str,
-        folder: &Folder,
-    ) -> Result<()>;
+        folder_id: &str,
```

Comment:
We really want to be consistent here. I would keep Folder for now and then move all endpoints to not using refed objects 🙏

Reply:
it was to be consistent with upsert_table,  which is already using individual fields  (discussed this  with @fontanierh )  .

Reply:
Right but upsert_table is non consistent with everything else, is it? :)

Reply:
Yes but it has fields that wouldn't make sense to pass

Reply:
So I don't think upsert table should change.

Reply:
Right after this PR is merged I'll take exploding the fields everywhere if that's OK ?

Reply:
I don't quite see the value really?

My take is the following if it's more redlines than greenlines then why not. If not I'm quite bearish. Long list of arguments are annoying really in general and the few cases where the ref to an object is inconvenient are really not that bad are they?

Reply:
In general having to specify many arguments vs an object that (in the case of nodes) will be constructuted automatically is much more error prone also.

I would suggest sticking to the standard while we figure that one out :pray:

Reply:
For Tables, we cannot pass the same object as input as the one we get as output. Some fields must not be populated at upsert time.
So the current setup cannot be the standard. If you want full consistency, the only way is refactor. I the goal to have consistency or not ?

Reply:
A different input object is also possible but more work. I really don't have a strong opinion on which setup we go with, but the one we currently have is by far the worst of all 3.

Reply:
I think the most idiomatic and clean solution (the one QDrant uses for example) is to have a specific owned object for params. I am happy to move forward with this solutio